src/linearmixedmodel.jl

```diff
@@ -776,9 +778,9 @@ This is the crucial step in evaluating the objective, given a new parameter valu
 function updateL!(m::LinearMixedModel{T}) where {T}
     A = m.A
     L = m.L
-    k = BlockArrays.nblocks(A, 2)
+    k = length(m.allterms)
     for j = 1:k                         # copy lower triangle of A to L
-        for i = j:BlockArrays.nblocks(A, 1)
+        for i = j:length(m.allterms)
```

Comment:
In the original, this was done for two different dimensions, but since it's lower triangular, I guess this doesn't matter and actually makes things more consistent/obvious.

Reply:
Is it better to write that as `j:k`?  I don't see an advantage in evaluating `length(m.allterms)` every iteration.  The compiler is probably smart enough to hoist that out of the loop but there is no reason to burden it.

Reply:
Probably -- the definition of `k` is only two lines ahead, so it's easy to see how `k` is defined and then it's obvious that both loops have the same upper bound.